tools/cpp/runfiles/runfiles_src.cc

```diff
@@ -111,6 +111,59 @@ bool ParseRepoMapping(const string& path,
 
 }  // namespace
 
+#if defined(__has_builtin)
+#if __has_builtin(__builtin_FILE)
+#define BAZEL_RUNFILES_GENERIC_HAS_BUILTIN_FILE
+#endif
+#endif
+#if defined(_MSC_VER)
+#if _MSC_VER >= 1926
+#define BAZEL_RUNFILES_MSVC_HAS_BUILTIN_FILE
+#endif
+#endif
+
+#if defined(BAZEL_RUNFILES_GENERIC_HAS_BUILTIN_FILE) ||                        \
+    defined(BAZEL_RUNFILES_MSVC_HAS_BUILTIN_FILE)
+std::string Runfiles::CurrentRepository(std::string file) {
+  std::string repo;
+  if (file[0] == 'e' && file[1] == 'x' && file[2] == 't' && file[3] == 'e' &&
```

Comment:
Could this use `startswith`? In any case it would need bounds checking as `file` may be too short.

Reply:
I don't know, but the tests should tell us.